include/seqan3/utility/views/type_reduce.hpp

```diff
@@ -53,8 +53,8 @@ class type_reduce_fn : public adaptor_base<type_reduce_fn>
             std::ranges::viewable_range<urng_t>,
             "The views::type_reduce adaptor can only be passed viewable_ranges, i.e. Views or &-to-non-View.");
 
-        // views are always passed as-is
-        if constexpr (std::ranges::view<std::remove_cvref_t<urng_t>>)
+        // string_view
+        if constexpr (is_type_specialisation_of_v<std::remove_cvref_t<urng_t>, std::basic_string_view>)
```

Comment:
can you elaborate on this change? It seems weird to replace an if clause for views with that of a very specific type (string_view).

In addition you are just removing tests

Reply:
The removed tests are stuff that's not actually in the standard, though we expected it to be.

Reply:
https://github.com/seqan/seqan3/blob/dbc7dd8e9886fb42007a48c0f71679bb42b7ecb9/include/seqan3/utility/views/slice.hpp#L66-L98